engine/verification/finder/engine.go

```diff
@@ -181,37 +172,34 @@ func (e *Engine) handleExecutionReceipt(originID flow.Identifier, receipt *flow.
 		defer span.Finish()
 	}
 	ctx = opentracing.ContextWithSpan(ctx, span)
-	childSpan, _ := e.tracer.StartSpanFromContext(ctx, trace.VERFindHandleExecutionReceipt)
-	defer childSpan.Finish()
-
-	receiptID := receipt.ID()
-	resultID := receipt.ExecutionResult.ID()
-
-	log := e.log.With().
-		Hex("origin_id", logging.ID(originID)).
-		Hex("receipt_id", logging.ID(receiptID)).
-		Hex("result_id", logging.ID(resultID)).Logger()
-	log.Info().
-		Msg("execution receipt arrived")
-
-	// monitoring: increases number of received execution receipts
-	e.metrics.OnExecutionReceiptReceived()
-
-	// caches receipt as a receipt data pack for further processing
-	rdp := &verification.ReceiptDataPack{
-		Receipt:  receipt,
-		OriginID: originID,
-		Ctx:      ctx,
-	}
 
-	ok = e.cachedReceipts.Add(rdp)
-	if !ok {
-		return fmt.Errorf("duplicate execution receipt. receipt_id: %x", logging.ID(receiptID))
-	}
-	log.Debug().
-		Msg("execution receipt successfully handled")
+	e.tracer.WithSpanFromContext(ctx, trace.VERFindHandleExecutionReceipt, func() {
```

Comment:
This looks great, thanks for the refactoring. 

I have one suggestion that could simplify it further.

Let's talk about the simple form of the pattern here, what we are doing is this

```
func (e *Engine) handleReceipt(origin, receipt) {
  e.tracerWithSpanFromContext(ctx, xxx, func() {
    ... business logic
  })
}
```

Instead, I think better to wrap with a function:

```
func (e *Engine) handleReceiptWithTracing(origin, receipt) {
  e.tracerWithSpanFromContext(ctx, xxx, func() {
    e.handleReceiptWithTracing(origin, receipt)
  })
}

func (e *Engine) handleReceipt(origin, receipt) {
  ... original business logic unchanged.
}

```

The benefit is that it's now easier to find the handleReceipt's logic, otherwise we also have to read through the tracing related code to find that business logic.

Reply:
thanks for the suggestion applied in then subsequent commits 👇